### ISSUE TITLE
[csharp] useDateTimeOffset is ignored and datetimes are always generated as "DateTime?"

### ISSUE THREAD

**Issue by rubms** [OpenAPITools/openapi-generator#1524]
##### Description

The `useDateTimeOffset` configuration property is ignored and datetimes are always generated as `DateTime?`. The expected behavior is to get `DateTimeOffset?` when  `useDateTimeOffset=true` and  `DateTime?` when  `useDateTimeOffset=false`.

This happens for both: operation parameters and object properties.

##### openapi-generator version

Error reproduced with openapi generator v3.3.3. The v3.3.2 version worked fine.

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.0
info:
  description: "Spec for testing the generation of DateTime properties"
  version: 1.0.0
  title: DateTime test case.
paths:
  /pets:
    get:
      operationId: getPets
      summary: Pets.
      description: |
        Get pets.
      parameters:
      - name: registrationDateTime
        in: query
        description: "Test datetime query parameter"
        required: false
        schema:
          type: string
          format: date-time
      tags:
      - Pets
      responses:
        200:
          description: Successful request.
```

##### Command line used for generation

`java -jar openapi-generator-cli-3.3.3.jar generate -g csharp -i test.yml -o test`

Additionally, the configuration property `useDateTimeOffset` is set to true:
```
{
	"useDateTimeOffset": "true"
}
```

##### Steps to reproduce

Generate a C# client using the csharp generator and specifying `useDateTimeOffset` to `true`.  Any date-time value will be generated as DateTime?, instead of DateTimeOffset?.

**Comment by rubms** [comment]
Created the https://github.com/OpenAPITools/openapi-generator/pull/1544 pull request that fixes the error.

**Comment by jimschubert** [comment]
@rubms I'm just getting caught up in issues and pull requests after moving into a new home. Thanks for submitting the issue and also the pull request!

Your contribution is available in v4.0.0-beta2 v4.0.0-beta, so I'm going to go ahead and close this. Thanks again.
